gitlab/riscv64.patch

```diff
@@ -11,30 +13,43 @@
  prepare() {
 +	# patch extconf.rb with sys("autoreconf -fi") to update config.guess and config.sub in extension msgpack of gem rbtrace
 +	# fix problem "config.guess: unable to guess system type"
-+	cd rbtrace/ext
-+	sed -Ei "/^([[:space:]]*)Dir.chdir\(dir\) do/a \sys(\"autoreconf -fi\")" extconf.rb
-+	cd ../
-+	gem build rbtrace.gemspec
-+	cd ../
++	sed -Ei "/^([[:space:]]*)Dir.chdir\(dir\) do/a \sys(\"autoreconf -fi\")" rbtrace/ext/extconf.rb
  	cd gitlab-foss
  
  	# GitLab tries to read its revision information from a file.
-@@ -124,8 +135,17 @@ build() {
+@@ -116,6 +123,11 @@ prepare() {
+ }
+ 
+ build() {
++	# build rbtrace, see prepare()
++	cd rbtrace
++	gem build rbtrace.gemspec
++	cd ..
++
+ 	cd gitlab-foss
+ 
+ 	# https://github.com/nodejs/node/issues/48444
+@@ -137,11 +149,22 @@ build() {
  	bundle config --local force_ruby_platform true # some native gems are not available for newer ruby
  	bundle config --local deployment true
  	bundle config --local without 'development test aws kerberos'
++	# override nokogiri build flags
++	bundle config --local build.nokogiri "--use-system-libraries --enable-shared --disable-static --with-cflags='-O2'"
```

Comment:
Wondering if this should be upstreamed to Arch Linux. 🤔

Reply:
Hello again. I don't know if upstreaming it to arch linux is up to the code of conduct, because according to investigation, x86 uses the prebuilt gem, so they are not compiling it natively. Adding this line to x86 build does no difference, the upside is that it will not affect the compiling behaviour, the downside, however, is that it is actually a meaningless line on x86 packing process.